src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -223,9 +223,7 @@ public void StartTagHelperWritingScope(HtmlEncoder encoder)
 
             // We need to replace the ViewContext's Writer to ensure that all content (including content written
             // from HTML helpers) is redirected.
-            var buffer = new ViewBuffer(BufferScope, Path);
-            var writer = new HtmlContentWrapperTextWriter(buffer, ViewContext.Writer.Encoding);
-            ViewContext.Writer = writer;
+            ViewContext.Writer = new ViewBufferTextWriter(buffer, ViewContext.Writer.Encoding);
```

Comment:
Users can set `ViewContext.Writer` too. Suggest we `throw` here if the property has an unexpected type where it's important. Bill Murray would be [chanting](https://www.youtube.com/watch?v=e9mf3Bypyk8) that "It Just Doesn't Matter" in a lot of this code but there's a few places where we shouldn't make the wrong assumption.